napari/notifications.py

```diff
@@ -0,0 +1,178 @@
+import os
+import sys
+import warnings
+from datetime import datetime
+from enum import auto
+from time import time
+from types import TracebackType
+from typing import Any, Callable, List, Sequence, Tuple, Type, Union
+
+from .utils.events import Event, EventEmitter
+from .utils.misc import StringEnum
+
+
+class NotificationSeverity(StringEnum):
+    """Severity levels for the notification dialog.  Along with icons for each."""
+
+    ERROR = auto()
+    WARNING = auto()
+    INFO = auto()
+    NONE = auto()
+
+    def as_icon(self):
+        return {
+            self.ERROR: "ⓧ",
+            self.WARNING: "⚠️",
+            self.INFO: "ⓘ",
+            self.NONE: "",
+        }[self]
+
+
+ActionSequence = Sequence[Tuple[str, Callable[[], None]]]
+
+
+# Event subclasses specific to the Canvas
```

Comment:
Why is this subclass specific to the `Canvas`? cc @tlambert03

Reply:
Lol. It's not.  Just a bad comment

Reply:
Ok, nice, we'll cut when we do final cleanup etc